vg/recorder/recorder.go

```diff
@@ -0,0 +1,458 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package recorder provides support for vector graphics serialization
+// it is intended to be used as a testing helper.
+package recorder
+
+import (
+	"fmt"
+	"image/color"
+	"runtime"
+
+	"github.com/gonum/plot/vg"
+)
+
+var _ vg.Canvas = (*Canvas)(nil)
+
+// Canvas implements vg.Canvas operation serialization.
+type Canvas struct {
```

Comment:
Not sure if this would ever be useful, but this could also have an underlying vg.Canvas to which it forwards calls. That way, it can log calls while actually drawing to another canvas too. If the underlying canvas is nil then it simply doesn't call its methods.